### ISSUE TITLE
Article about module support in web extensions

### ISSUE THREAD

**Issue by erosman** [mdn/content#45317]
### Preamble

The identical proposal applies to both [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) &  [import()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import). If required, I can split them into 2 topics.

### Details

The following relevant data would be beneficial to have in BCD (as finding the data requires going through many pages).

- Available in workers (already listed in both) :heavy_check_mark: 
- Available in background service worker (MV3)
- Available in background event page (MV3)
- Available in background scripts (MV2)
  Available by loading background script as module via a background page
- Available in content script

Also relevant to:

- [background](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/background)
- [content_scripts](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/content_scripts)

### Reference

[Support type: "module" (background service worker as a ES module)](https://bugzilla.mozilla.org/show_bug.cgi?id=1775574)
[Support type: "module" (event page, not background service worker)](https://bugzilla.mozilla.org/show_bug.cgi?id=1811443)
[Support loading content scripts as ES6 modules](https://bugzilla.mozilla.org/show_bug.cgi?id=1451545)
[Dynamic module import doesn't work in webextension content scripts](https://bugzilla.mozilla.org/show_bug.cgi?id=1536094)
[Dynamic import fails in content script in MV3 extension](https://bugzilla.mozilla.org/show_bug.cgi?id=1803950)

### How impactful do you think this enhancement will be?

Provide additional data as per aforementioned

**Comment by caugner** [comment]
IIUC the suggestion here is to add data for the availability of `import` and `import()` in different WebExtensions contexts.

To be honest, I'm not sure how we could capture this intersection of JavaScript and WebExtensions feature though, assuming this is relevant for extension developers.

@rebloor What do you think about adding this data? If this is relevant, where could it be added?

**Comment by rebloor** [comment]
@caugner I think that @erosman is suggesting we extend the BCD information along the lines of ["Available in service workers"](https://github.com/mdn/browser-compat-data/blob/b579ffc603c1098b61788583d09cc4c6f536124e/javascript/statements.json#L1833) the mechanics of which looked fairly straightforward. However, while I haven't started investigating yet, gathering the relevant browser support details may be more time-consuming. As I'm not sure how likely extension developers are to use the BCD as a reference, I'm inclined to see this as a nice-to-have.

**Comment by Rob--W** [comment]
I don't think that it makes sense to embed extension support in various contexts in the BCD of import statement and dynamic import(), since it is not relevant to most readers, and I doubt if interested readers would find the relevant docs.

If anything, an extension specific MDN article under webextensions/ about module support in various extension contexts would make more sense. With cross-references to content script and background script documentation.
